File: src/official/o2020/feb/bronze/billNyeFam/breedflip.py
# 2020 feb bronze
with open("breedflip.in") as read:
    cow_num = int(read.readline())
    needed = read.readline().strip()
    have_rn = read.readline().strip()
    assert len(needed) == len(have_rn) == cow_num

is_diff = [n != h for n, h in zip(needed, have_rn)]
consec_num = 0
for i in range(1, cow_num):
    consec_num += not is_diff[i] and is_diff[i - 1]
consec_num += is_diff[-1]

print(consec_num, file=open("breedflip.out", "w"))

File: src/official/o2020/feb/bronze/billNyeFam/BreedFlip.java
package official.o2020.feb.bronze.billNyeFam;

import java.io.*;

public class BreedFlip {
    public static void main(String[] args) throws IOException {
        BufferedReader read = new BufferedReader(new FileReader("breedflip.in"));
        int cowNum = Integer.parseInt(read.readLine());
        String needed = read.readLine();
        String haveRn = read.readLine();
        assert needed.length() == cowNum && haveRn.length() == cowNum;

        boolean[] isDiff = new boolean[cowNum];
        for (int c = 0; c < cowNum; c++) {
            isDiff[c] = needed.charAt(c) != haveRn.charAt(c);
        }

        int consecNum = 0;
        for (int i = 1; i < cowNum; i++) {
            if (!isDiff[i] && isDiff[i - 1]) {
                consecNum++;
            }
        }
        consecNum += isDiff[cowNum - 1] ? 1 : 0;

        PrintWriter written = new PrintWriter("breedflip.out");
        written.println(consecNum);
        written.close();
    }
}
